engine/src/execution_graph/logic_controllers/BatchWindowFunctionProcessing.cpp

```diff
@@ -0,0 +1,154 @@
+#include "BatchWindowFunctionProcessing.h"
+#include "execution_graph/logic_controllers/BlazingColumn.h"
+#include "taskflow/executor.h"
+#include "CodeTimer.h"
+
+#include <src/utilities/CommonOperations.h>
+#include "parser/expression_utils.hpp"
+#include <blazingdb/io/Util/StringUtil.h>
+
+#include <cudf/concatenate.hpp>
+#include <cudf/stream_compaction.hpp>
+#include "cudf/column/column_view.hpp"
+#include <cudf/rolling.hpp>
+#include <cudf/filling.hpp>
+#include <cudf/partitioning.hpp>
+#include <cudf/types.hpp>
+#include <cudf/copying.hpp>
+
+namespace ral {
+namespace batch {
+
+// BEGIN ComputeWindowKernel
+
+ComputeWindowKernel::ComputeWindowKernel(std::size_t kernel_id, const std::string & queryString,
+    std::shared_ptr<Context> context,
+    std::shared_ptr<ral::cache::graph> query_graph)
+    : kernel{kernel_id, queryString, context, kernel_type::ComputeWindowKernel} {
+    this->query_graph = query_graph;
+}
+
+// TODO: support for LAG(), LEAD(), currently looks like Calcite has an issue obtaining the optimized plan
+// TODO: Support for RANK() and DENSE_RANK() file an cudf feature/request
+// TODO: Support for first_value() and last_value() file an cudf feature/request
+std::unique_ptr<CudfColumn> ComputeWindowKernel::compute_column_from_window_function(cudf::table_view input_cudf_view, cudf::column_view input_col_view, std::size_t pos) {
+
+    std::unique_ptr<cudf::aggregation> window_function = get_window_aggregate(this->aggs_wind_func[pos]);
+    std::unique_ptr<CudfColumn> windowed_col;
+    std::vector<cudf::column_view> table_to_rolling;
+    table_to_rolling.push_back(input_cudf_view.column(this->column_indices_partitioned[0]));
```

Comment:
What happens if the window is partitioned by more than one column? I think here you are only considering when a window is only partitioned by one column. 
Also, this makes me realize we need e2e tests that partition by more than one column, do order by by more than one column and both.
Also please add e2e tests where there is a regular order by outside the window, to make sure that calcite and our own query parsing code does not get confused by more than one order by in the query